packages/zendesk-adapter/src/filters/custom_objects/custom_object_fields_order.ts

```diff
@@ -0,0 +1,161 @@
+  fields: {
+    [ORDER_FIELD]: { refType: new ListType(BuiltinTypes.NUMBER) },
+  },
+  path: [ZENDESK, elementsUtils.TYPES_PATH, CUSTOM_OBJECT_FIELD_ORDER_TYPE_NAME],
+})
+
+/**
+ Creates an order object for each custom object
+ The fields are returned from the api by order, so we save it to be able to properly reorder them on deploy
+ This is needed because two fields can have the same positions, and then be sorted by non multi-env fields
+ */
+const customObjectFieldsOrderFilter: FilterCreator = ({ client }) => ({
+  name: 'customObjectFieldsOrderFilter',
+  onFetch: async (elements: Element[]) => {
+    const customObjectFields = elements
+      .filter(isInstanceElement)
+      .filter(instance => instance.elemID.typeName === CUSTOM_OBJECT_FIELD_TYPE_NAME)
+    const customObjectsByFullName = _.keyBy(
+      elements.filter(isInstanceElement).filter(instance => instance.elemID.typeName === CUSTOM_OBJECT_TYPE_NAME),
+      element => element.elemID.getFullName()
+    )
+
+    const customObjectFieldsByParentName = _.groupBy(
+      customObjectFields.filter(field => getParents(field).length === 1),
+      field => getParent(field).elemID.getFullName()
+    )
+
+    if (!_.isEmpty(customObjectFieldsByParentName)) {
+      elements.push(customObjectFieldsOrderType)
+    }
+
+    Object.entries(customObjectFieldsByParentName).forEach(([parentName, fields]) => {
+      const parent = customObjectsByFullName[parentName]
+      const instanceName = `${parent.elemID.name}_fields_order`
+      const orderInstance = new InstanceElement(
+        instanceName,
+        customObjectFieldsOrderType,
+        {
+          [ORDER_FIELD]: fields.map(field => new ReferenceExpression(field.elemID, field)),
+        },
+        [ZENDESK, RECORDS_PATH, CUSTOM_OBJECT_FIELD_ORDER_TYPE_NAME, instanceName],
+        {
+          [CORE_ANNOTATIONS.PARENT]: parent
+            ? [new ReferenceExpression(parent.elemID, parent)]
+            : undefined,
+        }
+      )
+      elements.push(orderInstance)
+    })
+  },
+  deploy: async changes => {
+    const [relevantChanges, leftoverChanges] = _.partition(
+      changes,
+      change =>
+        isInstanceChange(change)
+        && isAdditionOrModificationChange(change)
+        && getChangeData(change).elemID.typeName === CUSTOM_OBJECT_FIELD_ORDER_TYPE_NAME
+    )
+
+    const deployResultsPromises = relevantChanges
+      .filter(isInstanceChange) // used to type check
+      .map(async change => {
+        const customObjectFieldOrder = getChangeData(change)
+        const parentKey = getParent(customObjectFieldOrder).value.key
+        if (parentKey === undefined) {
+          return {
+            change,
+            error: 'parent custom_object key is undefined',
+          }
+        }
+        const fieldsIds = customObjectFieldOrder.value[ORDER_FIELD]
+          .map((field: Value) => {
+            if (isResolvedReferenceExpression(field)) {
+              return field.value.value.id.toString()
```

Comment:
we need to think of a situation where id is not defined
```suggestion
              return field.value.value.id?.toString()
```